diff -burNp archfs-0.5.4/layout/support.c archfs-0.5.4-dud/layout/support.c
--- archfs-0.5.4/layout/support.c	2007-08-17 18:44:14.000000000 +0200
+++ archfs-0.5.4-dud/layout/support.c	2009-02-01 15:00:28.640399603 +0100
@@ -71,7 +71,7 @@ int unzip_revs(char *path){
                 gstrdel(extension);
             	if (gmstrcpy(&mirror, tmp_file, "/", entry->d_name, 0) == -1)
             		continue;
-            	if ((descriptor = open(mirror, O_WRONLY | O_CREAT)) == -1)
+            	if ((descriptor = open(mirror, O_WRONLY | O_CREAT, 0777)) == -1)
             		continue;
             	if (close(descriptor) == -1)
             		continue;
